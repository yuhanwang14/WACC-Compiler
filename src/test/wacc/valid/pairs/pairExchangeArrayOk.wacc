# Assignment is legal when assigning array (even of unknown type) in nested pair extraction
# Thanks to Kristina Zimina for catching this

# Output:

# Program:

begin
  pair(int, int) p = newpair(4, 5);
  pair(pair, int) q = newpair(p, 6);
  fst fst q = []
end
