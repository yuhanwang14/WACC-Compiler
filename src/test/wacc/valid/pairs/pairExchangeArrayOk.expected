 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
# Assignment is legal when assigning array (even of unknown type) in nested pair extraction
# Thanks to Kristina Zimina for catching this

# Output:

# Program:

begin
  pair(int, int) p = newpair(4, 5);
  pair(pair, int) q = newpair(p, 6);
  fst fst q = []
end
 